### PR TITLE
Application implementation

### PR DESCRIPTION
**Pull request by NotCampeur**

### CODE REVIEW

File: includes/graphics/linux/spk_window.hpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+		xcb_connection_t *_connection;
+		xcb_screen_t *_screen;
+		xcb_window_t _window;
+
+		Window(const std::wstring& p_title, spk::Vector2Int p_size);
+
+	public:
+		void setGeometry(spk::Vector2Int p_size);
+		void resize(spk::Vector2Int p_size);
```
Comment (EreliaCorp) [inline includes/graphics/linux/spk_window.hpp:25]: Pass the argument as const ref

File: src/threading/spk_worker_pool.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#include "threading/spk_worker_pool.hpp"
+
+namespace spk
+{
+    WorkerPool::WorkerPool(std::wstring p_poolName, size_t p_nbWorker)
```
Comment (EreliaCorp) [inline src/threading/spk_worker_pool.cpp:5]: Pass the argument as const ref

**Review by EreliaCorp — CHANGES_REQUESTED** [review]
Multiples requests are formulated :
- Reformating some files
- Changing parameter to const ref
- adding the "p_" to some argument
- etc etc

File: includes/application/modules/spk_abstract_module.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+#include "spk_system_define.hpp"
+
+namespace spk
+{
+	class AbstractModule
+	{
+	public:
+		virtual void update() = 0;
+	};
```
Comment (EreliaCorp) [inline includes/application/modules/spk_abstract_module.hpp:13]: Maybe this class isn't necessary, as we do not use it as base class, maybe we should remove it

File: includes/application/modules/windows/spk_API_module.hpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		spk::ThreadSafeQueue<SystemMessage> _mouseQueue;
+		spk::ThreadSafeQueue<SystemMessage> _keyboardQueue;
+
+		LRESULT _handleMessage(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam);
+
+	public:
+		APIModule();
+		~APIModule();
+
+		void update();
```
Comment (EreliaCorp) [inline includes/application/modules/windows/spk_API_module.hpp:27]: We should rename this method to something closer to what it does, like "pullMessage" or something like this

File: includes/application/modules/linux/spk_API_module.hpp [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	private:
+		spk::ThreadSafeQueue<SystemMessage> _windowQueue;
+		spk::ThreadSafeQueue<SystemMessage> _mouseQueue;
+		spk::ThreadSafeQueue<SystemMessage> _keyboardQueue;
+
+	public:
+		APIModule();
+		~APIModule();
+
+		void update();
```
Comment (EreliaCorp) [inline includes/application/modules/linux/spk_API_module.hpp:22]: We should rename this method to something closer to what it does, like "pullMessage" or something like this

File: includes/application/concrete/windows/spk_graphical_application.hpp [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+		spk::MouseModule *_mouseModule;
+		spk::KeyboardModule *_keyboardModule;
+		spk::SystemModule *_systemModule;
+
+		spk::WidgetModule* _widgetModule;
+
+	protected:
+		void setupJobs()
+		{
+			addJob([&](){ _APIModule->update(); });
```
[the remark this answers is not in the document]
Comment (NotCampeur) [reply includes/application/concrete/windows/spk_graphical_application.hpp:29]: postponed.

**Review by EreliaCorp — APPROVED** [review]

**Review by EreliaCorp — APPROVED** [review]
Review okay

**Pull request merged**